core/src/main/scala/org/apache/spark/scheduler/SchedulerBackend.scala

```diff
@@ -53,6 +54,8 @@ private[spark] trait SchedulerBackend {
       reason: String): Unit =
     throw new UnsupportedOperationException
 
+  def getTaskThreadDump(taskId: Long, executorId: String): Option[ThreadStackTrace]
```

Comment:
Do we want to specify a timeout here (and propagate it) ? In case executor is severely loaded.
This is currently not being done for executor thread dumps as well though ...

Reply:
Currently, it depends on the `spark.rpc.askTimeout/spark.network.timeout`, I am not sure if we need a separate timeout key for some particular case like here